### PR TITLE
os/: add ST7701SN LCD support

### PR DESCRIPTION
**Pull request by namanjain7**
add ST7701SN LCD support. The LCD supports MIPI.

### CODE REVIEW

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 1]
```diff
@@ -441,11 +441,10 @@ void board_initialize(void)
 	rtl8730_st7789_initialize();
 #endif
 
-#if defined(CONFIG_LCD_ST7785) || defined(CONFIG_LCD_ST7701)
-	rtl8730e_lcdc_initialize();
+#if defined(CONFIG_LCD_ST7785) | defined(CONFIG_LCD_ST7701) | defined(CONFIG_LCD_ST7701SN)
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:444]: Maybe `||`, not `|`.

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 2]
```diff
@@ -441,11 +441,10 @@ void board_initialize(void)
...
+        rtl8730e_lcdc_initialize();
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:445]: Let's revert this line change.

File: os/drivers/lcd/Kconfig [hunk 3]
```diff
@@ -119,6 +119,11 @@ config LCD_ST7701
 	---help---
 		Enable LCD Driver, ST7701 with mipi.
 
+config LCD_ST7701SN
```
Comment (sunghan-chang) [inline os/drivers/lcd/Kconfig:122]: ST7701 and ST7701SN are different?

File: os/include/tinyara/lcd/mipi_lcd.h [hunk 4]
```diff
@@ -35,7 +35,7 @@
 
 #if defined(CONFIG_LCD_ST7785)
 #define INIT_CMD_SIZE 28
-#elif defined(CONFIG_LCD_ST7701)
+#elif defined(CONFIG_LCD_ST7701) | defined( CONFIG_LCD_ST7701SN)
```
Comment (sunghan-chang) [inline os/include/tinyara/lcd/mipi_lcd.h:38]: same as previous

**Comment by edwakuwaku** [comment]
@namanjain7 , I think ST7701SN supports MIPI 2-lane. Some init code needs to be changed for better performance.

**Comment by sunghan-chang** [comment]
@namanjain7 Could you update the PR for the comments?

**Comment by namanjain7** [comment]
> @namanjain7 Could you update the PR for the comments?

Yes sir, this PR isn't cleaned. It's just a code which is working with 4.3 inch LCD but is not ready to be merged. We are cleaning code right now, and also solving issues. We will push the cleaned code in this PR today.

File: os/drivers/lcd/Kconfig (hunk 3 above)
Comment (namanjain7) [reply os/drivers/lcd/Kconfig:122]: st7701 -> 3.9 inch LCD
st7701sn -> 4.3 inch LCD

File: os/include/tinyara/lcd/mipi_lcd.h [hunk 5]
```diff
@@ -35,7 +35,7 @@
 
 #if defined(CONFIG_LCD_ST7785)
 #define INIT_CMD_SIZE 28
-#elif defined(CONFIG_LCD_ST7701)
+#elif defined(CONFIG_LCD_ST7701) || defined( CONFIG_LCD_ST7701SN)
```
Comment (sunghan-chang) [inline os/include/tinyara/lcd/mipi_lcd.h:38]: Minor comment. Let's remove a space between `d( C`.
defined( CONFIG...)
->
defined(CONFIG...)

**Comment by namanjain7** [comment]
@edwakuwaku sure, I am going to add proper comments for init sequence in new PR.

**Pull request merged**